src/array.ts

```diff
@@ -66,11 +83,21 @@ export function array<A>(
         return propagateFail(failOrThrow, item, v, i)
       }
 
-      if (!isPure) {
+      if (!meta.isPure) {
         res[i] = item
       }
     }
 
     return res
-  }, isPure)
+  }, meta)
+}
+
+export function toSchema(
+  runtype: Runtype<any[]>,
+  runtypeToSchema: (runtype: Runtype<any>) => string,
+): string {
+  const meta: Meta = (runtype as any).meta
```

Comment:
Yay! Finally in a good mood to look at this :grin: 

First question/suggestion: I would use a getter here instead of manually writing `const meta: Meta = (runtype as any).meta` again and again. sth like an internal `function getMeta<M>(rt: Runtype<any>): M` maybe even with a simple check that `.meta` exists and is an object.